pkg/operation/botanist/component/vpa/recommender.go

```diff
@@ -244,3 +262,17 @@ func (v *vpa) computeRecommenderCommands() []string {
 	}
 	return out
 }
+
+// recommenderLegacyResourceConfigs() returns a list of resources which were part of the recommender setup in the past
+// but no longer are. It is used to ensure the removal of these, now unnecessary, resources.
+// TODO: Remove this function in a future release, after the respective resources have been removed from all landscapes
+func (v *vpa) recommenderLegacyResourceConfigs() component.ResourceConfigs {
+	if v.values.ClusterType != component.ClusterTypeSeed {
+		// The recommender vpa resource was removed only for seeds. It is still in use in shoot clusters control planes
+		return nil
+	}
+
+	return component.ResourceConfigs{
+		{Obj: v.emptyVerticalPodAutoscaler(recommender), Class: component.Runtime, MutateFn: func() {}},
+	}
+}
```

Comment:
This should not be needed, VPA is managed via `ManagedResource`, so GRM should clean this up automatically.